app/models/host/managed.rb

```diff
@@ -604,9 +604,7 @@ def hash_clone(value)
 
   def set_hostgroup_defaults
     return unless hostgroup
-    assign_hostgroup_attributes(%w{domain_id compute_profile_id})
```

Comment:
It is handled by `#apply_inherited_attributes`. Look at: https://github.com/theforeman/foreman/blob/develop/app/models/host/managed.rb#L568